jena-fuseki2/jena-fuseki-core/src/main/java/org/apache/jena/fuseki/servlets/prefixes/PrefixesRDF.java

```diff
@@ -0,0 +1,220 @@
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.atomic.AtomicBoolean;
+
+import org.apache.jena.query.*;
+import org.apache.jena.sparql.core.DatasetGraph;
+import org.apache.jena.sparql.core.Transactional;
+import org.apache.jena.sparql.exec.QueryExec;
+import org.apache.jena.sparql.exec.RowSet;
+import org.apache.jena.sparql.exec.UpdateExec;
+import org.apache.jena.tdb2.DatabaseMgr;
+
+/**
+ * The prefix-URI mappings are represented in blank nodes of type Prefix, with prefixName and prefixURI attributes.
+ * The prefixName and prefixURI of the same node are a distinct prefix-URI mapping.
+ *
+ * PREFIX prefixes: <http://jena.apache.org/prefixes#>
+ * PREFIX rdf:      <http://www.w3.org/1999/02/22-rdf-syntax-ns#>
+ * PREFIX xsd:      <http://www.w3.org/2001/XMLSchema#>
+ *
+ * [] rdf:type prefixes:Prefix ;
+ *    prefixes:prefixName  "test" ;
+ *    prefixes:prefixURI   "http://example/testPrefix#"^^xsd:anyURI
+ *    .
+ *
+ * [ rdf:type prefixes:Prefix ;
+ *   prefixes:prefixName   "ns" ;
+ *   prefixes:prefixURI    "http://example/namespace#"^^xsd:anyURI
+ * ]
+ */
+
+public class PrefixesRDF implements PrefixesAccess {
+
+    DatasetGraph dataset = DatabaseMgr.createDatasetGraph();
+    public void setDataset() {
+
+        dataset.begin(ReadWrite.WRITE) ;
+        try {
+
+            String update1 = """
+                   PREFIX prefixes: <http://jena.apache.org/prefixes#>
+                   PREFIX rdf:      <http://www.w3.org/1999/02/22-rdf-syntax-ns#>
+                   PREFIX xsd:      <http://www.w3.org/2001/XMLSchema#>
+                   
+                   INSERT DATA {
+                         [] rdf:type prefixes:Prefix ;
+                            prefixes:prefixName  "prefix1" ;
+                            prefixes:prefixURI   "http://www.localhost.org/uri1#"^^xsd:anyURI
+                   }""";
+            UpdateExec.dataset(dataset).update(update1).execute();
+
+            String update2 = """
+                    PREFIX prefixes: <http://jena.apache.org/prefixes#>
+                    PREFIX rdf:      <http://www.w3.org/1999/02/22-rdf-syntax-ns#>
+                    PREFIX xsd:      <http://www.w3.org/2001/XMLSchema#>
+                    
+                    INSERT DATA {
+                          [] rdf:type prefixes:Prefix ;
+                             prefixes:prefixName  "prefix2" ;
+                             prefixes:prefixURI   "http://www.localhost.org/uri2#"^^xsd:anyURI
+                    }""";
+            UpdateExec.dataset(dataset).update(update2).execute();
+
+            dataset.commit();
+        } catch (Throwable th) {
+            dataset.abort();
+            throw(th);
+        } finally {
+            dataset.end() ;
+        }
+
+    }
+
+    public Transactional transactional() { return dataset; }
+
+    @Override
+    public Optional<String> fetchURI(String prefix) {
+
+        String query = "PREFIX prefixes: <http://jena.apache.org/prefixes#> " +
```

Comment:
Could be better to use a parameterised string here i.e. https://jena.apache.org/documentation/query/parameterized-sparql-strings.html